### PR TITLE
Add ! operator to PinValue

### PR DESCRIPTION
**Pull request by pgrawehr**
Split away some parts of larger PRs, so that hopefully we can merge quicker.

### CODE REVIEW

File: src/System.Device.Gpio/System/Device/Gpio/PinValue.cs [hunk 1]
```diff
@@ -82,6 +82,21 @@ public override bool Equals(object? obj)
...
+        /// <returns>High, when the input is low, low when the input is high</returns>
+        public static PinValue operator !(PinValue v)
+        {
+            if (v._value == 0)
+            {
+                return High;
+            }
+
+            return Low;
+        }
```
Comment (krwq) [inline src/System.Device.Gpio/System/Device/Gpio/PinValue.cs:98]:
```suggestion
        public static PinValue operator !(PinValue v) => (v._value == 0) ? High : Low;
```

**Review by krwq — APPROVED** [review]
LGTM

**Comment by krwq** [comment]
Thanks, @joperezr @Ellerbach since this is technically public API we require second approver, PTAL

**Review by Ellerbach — APPROVED** [review]
I prefer the short form

File: src/System.Device.Gpio/System/Device/Gpio/PinValue.cs (hunk 1 above)
Comment (pgrawehr) [reply src/System.Device.Gpio/System/Device/Gpio/PinValue.cs:98]: Personally, I prefer the long form (especially, if it includes logic, like here). Are there any recommendations on this?

**Review by joperezr — APPROVED** [review]
The change itself LGTM. Seems like we currently don't have unit tests for this struct but we should definitely add coverage for it. If you could add some unit tests for them on src/System.Device.Gpio.Tests project that would be ideal, if not please log an issue so that we add coverage soon.

**Comment by pgrawehr** [comment]
@joperezr : Just setting up a bunch of unit tests. I'll include this as well.

**Pull request merged**